thriftpy2/thrift.py

```diff
@@ -68,9 +68,22 @@ def __init__(self):
             pass
         return __init__
 
+
     varnames, defaults = zip(*spec)
```

Comment:
@ethe This function still returning the same function as before, just that init generated would have positional arguments now cause I didn't change `defaults` variable which is being passed to `types.FunctionType` while returning. What do you think?

Reply:
Unless we change defaults, I don't think this commit makes any difference.

```
In [4]: ab = thriftpy2.load("tests/base.thrift")

In [5]: ab.Hello.__init__
Out[5]: <unbound method Hello.__init__>

In [6]: ab.Hello.__init__??
Signature: ab.Hello.__init__(self, greet=None, name=None)
Source:   
def __init__(self, greet, name=None):
    self.name = name
File:      ~/Work/Personal/thriftpy2/<generated Hello.__init__>
Type:      instancemethod

In [9]: ab.Hello()
Out[9]: Hello(name=None, greet=None)
```
This should raise error, right?

Reply:
Yes, I think raising error here should be better, maybe we can remove required arguments from defaults.

Reply:
Removing required arguments from defaults breaking lots of tests. Look at new change and breaking test cases. I guess we can keep the signature intact and add `ValueError` statements inside __init__ if args are None and if required.

Reply:
@ethe Look at this commit: https://github.com/Thriftpy/thriftpy2/pull/132/commits/5617a5520686c767f96848a89c23704c3f599a04